eole/predict/inference.py

```diff
@@ -674,8 +677,7 @@ def _decode_and_generate(
         else:
             src_pad_mask = None
         tgt_pad_mask = decoder_in.eq(self._tgt_pad_idx).unsqueeze(1)  # [B, 1, T_tgt]
-
-        emb = self.model.tgt_emb(decoder_in, step=step)
+        position_embeddings = self.model.rope.update(decoder_in.size(1), step=step)
         dec_out, dec_attn = self.model.decoder(
```

Comment:
Why aren't we calling `_clear_cache` here (unlike in generator.py)?

Reply:
you're mistaken. we don't disable the cache when step decoding. We disable it when using the one-pass for estimation